File: low.cs
private int low(int[] array, int size)
{
	int i;
	int j;
	int k;
	int temp;
	for (i = 1; i < size; i *= 2)
	{
		for (j = 0; j < size - i; j += 2 * i)
		{
			k = j + i;
			if (k > size - 1)
			{
				k = size - 1;
			}
			for (temp = j; temp < k; temp++)
			{
				if (array[temp] > array[k])
				{
					int temp2 = array[temp];
					array[temp] = array[k];
					array[k] = temp2;
				}
			}
		}
	}
	return 0;
}

File: low.vb
Private Function low(ByVal array As Integer(), ByVal size As Integer) As Integer
    Dim i As Integer
    Dim j As Integer
    Dim k As Integer
    Dim temp As Integer
    i = 1

    While i < size
        j = 0

        While j < size - i
            k = j + i
            If k > size - 1 Then
                k = size - 1
            End If
            For temp = j To k - 1
                If array(temp) > array(k) Then
                    Dim temp2 = array(temp)
                    array(temp) = array(k)
                    array(k) = temp2
                End If
            Next

            j += 2 * i
        End While

        i *= 2
    End While
    Return 0
End Function
